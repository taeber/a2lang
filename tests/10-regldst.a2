asm {*
	ORG $800
	JSR main
	JMP $3D0
}

use [
    INIT: sub @ $FB2F,  ; AKA TEXT
    HOME: sub @ $FC58,
    ; COUT: sub <- [ch: char @ A] @ $FDED,
    CROUT: sub @ $FD8E,
    PRNTYX: sub <- [val: word @ YX] @ $F940,
]

let main = sub {
    INIT()
    HOME()

    PRNTYX($BEAD)
    CROUT()
}

