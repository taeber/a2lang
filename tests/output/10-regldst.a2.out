*
	ORG $800
	JSR main
	JMP $3D0
INIT	EQU $FB2F
HOME	EQU $FC58
CROUT	EQU $FD8E
PRNTYX	EQU $F940
main	JSR INIT
	JSR HOME
* COPYWW @YX #$BE,#$AD
	LDX #$AD
	LDY #$BE
	JSR PRNTYX
	JMP CROUT
